pkg/controller/authentication/authentication_controller.go

```diff
@@ -0,0 +1,269 @@
+	}
+
+	for _, namespace := range []string{render.OperatorNamespace(), render.DexNamespace} {
+		for _, secretName := range []string{
+			render.DexTLSSecretName, render.OIDCSecretName, render.OpenshiftSecretName, render.DexObjectName,
+		} {
+			if err = utils.AddSecretsWatch(c, secretName, namespace); err != nil {
+				return fmt.Errorf("%s failed to watch the secret '%s' in '%s' namespace: %w", ControllerName, secretName, namespace, err)
+			}
+		}
+	}
+
+	return nil
+}
+
+// blank assignment to verify that ReconcileAuthentication implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileAuthentication{}
+
+// ReconcileAuthentication reconciles an Authentication object
+type ReconcileAuthentication struct {
+	client   client.Client
+	scheme   *runtime.Scheme
+	provider oprv1.Provider
+	status   status.StatusManager
+}
+
+// Reconciles the cluster state with the Authentication object that is found in the cluster.
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileAuthentication) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling ", "controller", ControllerName)
+
+	ctx := context.Background()
+
+	// Fetch the Authentication spec. If present, we deploy dex in the cluster
+	authentication, err := utils.GetAuthentication(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.OnCRNotFound()
+		}
+		return reconcile.Result{}, err
+	}
+	r.status.OnCRFound()
+	reqLogger.V(2).Info("Loaded config", "config", authentication)
+
+	// Query for the installation object.
+	network, err := installation.GetInstallation(context.Background(), r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			log.Error(err, "Installation not found")
+			r.status.SetDegraded("Installation not found", err.Error())
+			return reconcile.Result{}, err
+		}
+		log.Error(err, "Error querying installation")
+		r.status.SetDegraded("Error querying installation", err.Error())
+		return reconcile.Result{}, err
+	}
+	if network.Status.Variant != oprv1.TigeraSecureEnterprise {
+		log.Error(err, fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise))
+		r.status.SetDegraded(fmt.Sprintf("Waiting for network to be %s", oprv1.TigeraSecureEnterprise), "")
+		return reconcile.Result{}, nil
+	}
+
+	// Make sure authentication and managementclusterconnection are not present at the same time.
+	_, err = utils.GetManagementClusterConnection(ctx, r.client)
+	if err == nil {
+		log.Error(err, "Having Authentication CR and ManagementCluster CR at the same time is not supported")
+		r.status.SetDegraded("Having Authentication CR and ManagementCluster CR at the same time is not supported", err.Error())
```

Comment:
This is a user facing message, so:
- We shouldn't abbreviate CR
- We should spell out the full API names (ManagementClusterConnection rather than ManagementCluster).

e.g., "Only one of Authentication and ManagementClusterConnection may be specified"

Reply:
That was a mistake. Thanks for catching it.